stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,199 @@
+    ) -> tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod[_Context], str]]]: ...
+    def no_match(
+        self, context: _Context, transitions: tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod[_Context], str]]]
+    ) -> _TransitionResult[_Context]: ...
+    def bof(self, context: _Context) -> tuple[list[str], list[str]]: ...
+    def eof(self, context: _Context) -> list[str]: ...
+    def nop(self, match: Match[str], context: _Context, next_state: str) -> _TransitionResult[_Context]: ...
+
+class StateMachineWS(StateMachine[_Context]):
+    def get_indented(self, until_blank: bool = False, strip_indent: bool = True) -> tuple[StringList, int, int, bool]: ...
+    def get_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True
+    ) -> tuple[list[str], int, bool]: ...
+    def get_first_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True, strip_top: bool = True
+    ) -> tuple[list[str], int, int, bool]: ...
+
+class StateWS(State[_Context]):
+    indent_sm: type[StateMachine[_Context]] | None
+    indent_sm_kwargs: dict[str, Any] | None
+    known_indent_sm: type[StateMachine[_Context]] | None
+    known_indent_sm_kwargs: dict[str, Any] | None
+    ws_patterns: dict[str, Pattern[str]]
+    ws_initial_transitions: Sequence[str]
+    def __init__(self, state_machine: StateMachine[_Context], debug: bool = False) -> None: ...
+    def add_initial_transitions(self) -> None: ...
+    def blank(self, match: Match[str], context: _Context, next_state: str) -> _TransitionResult[_Context]: ...
+    def indent(self, match: Match[str], context: _Context, next_state: str) -> _TransitionResult[_Context]: ...
+    def known_indent(self, match: Match[str], context: _Context, next_state: str) -> _TransitionResult[_Context]: ...
+    def first_known_indent(self, match: Match[str], context: _Context, next_state: str) -> _TransitionResult[_Context]: ...
+
+class _SearchOverride:
+    def match(self, pattern: Pattern[str]) -> Match[str]: ...
+
+class SearchStateMachine(_SearchOverride, StateMachine[_Context]): ...
+class SearchStateMachineWS(_SearchOverride, StateMachineWS[_Context]): ...
+
+class ViewList(Generic[_T]):
+    data: list[_T]
+    items: list[tuple[str, int]]
+    parent: Self
+    parent_offset: int
+    def __init__(
+        self,
+        initlist: Self | Sequence[_T] | None = None,
+        source: str | None = None,
+        items: list[tuple[str, int]] | None = None,
+        parent: Self | None = None,
+        parent_offset: int | None = None,
+    ) -> None: ...
+    def __lt__(self, other: Any) -> bool: ...
+    def __le__(self, other: Any) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    def __gt__(self, other: Any) -> bool: ...
+    def __ge__(self, other: Any) -> bool: ...
+    def __contains__(self, item: Self) -> bool: ...
```

Comment:
`item` should either be `_T` or `object`. The implementation does just `item in self.data`, where `self.data` is `list[_T]`.